app/javascript/components/charts/am-chart-pie/AmChartPie.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="am-chart--pie">
+    <div class="chart__chart">
+      <div 
+        class="chart__svg"
+        id="chart-pie"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import * as am4core from "@amcharts/amcharts4/core"
+import * as am4charts from "@amcharts/amcharts4/charts"
+
+export default {
+  name: 'AmChartPie',
+
+  props: {
+    dataset: {
+      required: true,
+      type: Array // [{ title: String, value: Number }]
+    },
+    doughnut: {
+      default: false,
+      type: Boolean
+    },
+    spacers: {
+      default: false,
+      type: Boolean
+    }
+  },
+
+  mounted () {
+    this.createChart()
+  },
+
+  methods: {
+    createChart () {
+      const chart = am4core.create('chart-pie', am4charts.PieChart);
+      chart.data = this.dataset
+      chart.radius = am4core.percent(90)
+
+      if(this.doughnut) {
+        chart.innerRadius = am4core.percent(50)
+      }
+
+      const pieSeries = chart.series.push(new am4charts.PieSeries())
+      pieSeries.dataFields.id = 'id'
+      pieSeries.dataFields.category = 'title'
+      pieSeries.dataFields.value = 'value'
+      
+      if(this.spacers) {
+        pieSeries.slices.template.stroke = am4core.color('#ffffff')
+        pieSeries.slices.template.strokeWidth = 2
+        pieSeries.slices.template.strokeOpacity = 1
+      }
+
+      pieSeries.labels.template.disabled = true
+      pieSeries.ticks.template.disabled = true
+      //must match $theme-chart in settings.scss
+      pieSeries.colors.list = [
+        am4core.color('#64BAD9'),
+        am4core.color('#A54897'),
+        am4core.color('#65C9B2'),
+        am4core.color('#5F81CB'),
+        am4core.color('#FAA51B'),
+        am4core.color('#EF5F6C'),
+        am4core.color('#151617'),
+        am4core.color('#71A22B'),
+        am4core.color('#F5F58A'),
+        am4core.color('#EF266C'),
+        am4core.color('#1A4D9F'),
+        am4core.color('#E57133')
+      ]
```

Comment:
I was wondering whether you could define this list in a mixin or put it in the global scope somehow if it's going to be constant, then I came across this which was quite interesting: https://til.hashrocket.com/posts/sxbrscjuqu-share-scss-variables-with-javascript